RevokeConfirmation fails and always tries to run when Confirmation status is changed to Unconfirmed

**Describe the bug**
RevokeConfirmation fails when the confirmation status is reset to Unconfirmed.
The jira API will return an empty String for the changelog item, when a previously confirmed Ticket is set to unconfirmed again.
This causes the module to see an empty String in the change log entry, while the value on the ticket is `Unconfirmed`.

Because of this, the module tries to set the confirmation staus to empty String, which will fail with the following exception:
```
net.rcarz.jiraclient.JiraException: Failed to transition issue MCPE-69864
    at net.rcarz.jiraclient.Issue$FluentTransition.realExecute(Issue.java:484)
    at net.rcarz.jiraclient.Issue$FluentTransition.execute(Issue.java:518)
    at io.github.mojira.arisa.infrastructure.JiraKt$updateConfirmation$1$1.invokeSuspend(Jira.kt:45)
    at io.github.mojira.arisa.infrastructure.JiraKt$updateConfirmation$1$1.invoke(Jira.kt)
    at arrow.core.Either$Companion.catch(Either.kt:862)
    at arrow.core.Either$Companion.catch(Either.kt:858)
    at io.github.mojira.arisa.infrastructure.JiraKt$updateConfirmation$1.invokeSuspend(Jira.kt:38)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
    at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:271)
    at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:79)
    at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:54)
    at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
    at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:36)
    at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
    at io.github.mojira.arisa.infrastructure.JiraKt.updateConfirmation(Jira.kt:37)
    at io.github.mojira.arisa.ModuleExecutor$execute$13$1$2.invoke(ModuleExecutor.kt:337)
    at io.github.mojira.arisa.ModuleExecutor$execute$13$1$2.invoke(ModuleExecutor.kt:56)
    at arrow.syntax.function.PartialsKt$partially1$3.invoke(partials.kt:777)
    at arrow.syntax.function.PartialsKt$partially1$2.invoke(partials.kt:773)
    at io.github.mojira.arisa.modules.RevokeConfirmationModule$invoke$$inlined$with$lambda$1.invokeSuspend(RevokeConfirmationModule.kt:31)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at arrow.typeclasses.MonadContinuation$bind$$inlined$suspendCoroutineUninterceptedOrReturn$lambda$1.invoke(MonadContinuations.kt:36)
    at arrow.typeclasses.MonadContinuation$bind$$inlined$suspendCoroutineUninterceptedOrReturn$lambda$1.invoke(MonadContinuations.kt:15)
    at arrow.core.extensions.EitherMonad$flatMap$1.invoke(either.kt:123)
    at arrow.core.extensions.EitherMonad$flatMap$1.invoke(either.kt:115)
    at arrow.core.EitherKt.flatMap(Either.kt:881)
    at arrow.core.extensions.EitherMonad$DefaultImpls.flatMap(either.kt:123)
    at arrow.core.extensions.either.monad.EitherMonadKt$monad_singleton$1.flatMap(EitherMonad.kt:21)
    at arrow.core.extensions.either.monad.EitherMonadKt$monad_singleton$1.flatMap(EitherMonad.kt:21)
    at arrow.typeclasses.MonadContinuation.flatMap(MonadContinuations.kt)
    at arrow.typeclasses.MonadContinuation.bind$suspendImpl(MonadContinuations.kt:34)
    at arrow.typeclasses.MonadContinuation.bind(MonadContinuations.kt)
    at io.github.mojira.arisa.modules.RevokeConfirmationModule$invoke$$inlined$with$lambda$1.invokeSuspend(RevokeConfirmationModule.kt:30)
    at io.github.mojira.arisa.modules.RevokeConfirmationModule$invoke$$inlined$with$lambda$1.invoke(RevokeConfirmationModule.kt)
    at arrow.typeclasses.MonadFx$monad$wrapReturn$1.invokeSuspend(Monad.kt:108)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlin.coroutines.ContinuationKt.startCoroutine(Continuation.kt:128)
    at arrow.typeclasses.MonadFx$DefaultImpls.monad(Monad.kt:109)
    at arrow.core.extensions.EitherMonadFx.monad(either.kt:136)
    at arrow.core.extensions.EitherMonadFx.monad(either.kt:133)
    at arrow.core.extensions.EitherKt.fx(either.kt:235)
    at io.github.mojira.arisa.modules.RevokeConfirmationModule.invoke(RevokeConfirmationModule.kt:23)
    at io.github.mojira.arisa.ModuleExecutor$execute$13$1.invoke(ModuleExecutor.kt:320)
    at io.github.mojira.arisa.ModuleExecutor$execute$13$1.invoke(ModuleExecutor.kt:56)
    at io.github.mojira.arisa.ModuleExecutor.tryExecuteModule(ModuleExecutor.kt:466)
    at io.github.mojira.arisa.ModuleExecutor.access$tryExecuteModule(ModuleExecutor.kt:56)
    at io.github.mojira.arisa.ModuleExecutor$execute$13.invoke(ModuleExecutor.kt:318)
    at io.github.mojira.arisa.ModuleExecutor$execute$13.invoke(ModuleExecutor.kt:56)
    at io.github.mojira.arisa.ModuleExecutor.executeModule(ModuleExecutor.kt:402)
    at io.github.mojira.arisa.ModuleExecutor.access$executeModule(ModuleExecutor.kt:56)
    at io.github.mojira.arisa.ModuleExecutor$execute$exec$1.invoke(ModuleExecutor.kt:94)
    at io.github.mojira.arisa.ModuleExecutor$execute$exec$1.invoke(ModuleExecutor.kt:56)
    at arrow.syntax.function.PartialsKt$partially2$5.invoke(partials.kt:803)
    at arrow.syntax.function.PartialsKt$partially2$4.invoke(partials.kt:793)
    at arrow.syntax.function.PartialsKt$partially2$3.invoke(partials.kt:785)
    at arrow.syntax.function.PartialsKt$partially2$2.invoke(partials.kt:779)
    at io.github.mojira.arisa.ModuleExecutor.execute(ModuleExecutor.kt:317)
    at io.github.mojira.arisa.ArisaMainKt.main(ArisaMain.kt:51)
    at io.github.mojira.arisa.ArisaMainKt.main(ArisaMain.kt)
Caused by: net.rcarz.jiraclient.RestException: 400 : {"errorMessages":[],"errors":{"customfield_10500":"Option id 'null' is not valid"}}
    at net.rcarz.jiraclient.RestClient.request(RestClient.java:171)
    at net.rcarz.jiraclient.RestClient.request(RestClient.java:193)
    at net.rcarz.jiraclient.RestClient.request(RestClient.java:238)
    at net.rcarz.jiraclient.RestClient.post(RestClient.java:328)
    at net.rcarz.jiraclient.RestClient.post(RestClient.java:372)
    at net.rcarz.jiraclient.Issue$FluentTransition.realExecute(Issue.java:482)
     ... 59 common frames omitted 
```

--- a/src/main/kotlin/io/github/mojira/arisa/modules/RevokeConfirmationModule.kt
+++ b/src/main/kotlin/io/github/mojira/arisa/modules/RevokeConfirmationModule.kt
@@ -25,7 +25,7 @@ class RevokeConfirmationModule : Module<RevokeConfirmationModule.Request> {
                 .filter(::isConfirmationChange)
                 .filter(::changedByVolunteer)
                 .lastOrNull()
-                ?.newValue ?: "Unconfirmed"
+                ?.newValue.getOrDefault("Unconfirmed")
 
             assertNotEquals(confirmationStatus, volunteerConfirmation).bind()
             setConfirmationStatus(volunteerConfirmation).toFailedModuleEither().bind()
@@ -42,4 +42,10 @@ class RevokeConfirmationModule : Module<RevokeConfirmationModule.Request> {
         .created
         .plus(1, ChronoUnit.DAYS)
         .isAfter(Instant.now())
+
+    private fun String?.getOrDefault(default: String) =
+        if (isNullOrBlank())
+            default
+        else
+            this!!
 }

--- a/src/main/kotlin/io/github/mojira/arisa/modules/RevokeConfirmationModule.kt
+++ b/src/main/kotlin/io/github/mojira/arisa/modules/RevokeConfirmationModule.kt
@@ -27,7 +27,7 @@ class RevokeConfirmationModule : Module<RevokeConfirmationModule.Request> {
                 .lastOrNull()
                 ?.newValue.getOrDefault("Unconfirmed")
 
-            assertNotEquals(confirmationStatus, volunteerConfirmation).bind()
+            assertNotEquals(confirmationStatus.getOrDefault("Unconfirmed"), volunteerConfirmation).bind()
             setConfirmationStatus(volunteerConfirmation).toFailedModuleEither().bind()
         }
     }

--- a/src/test/kotlin/io/github/mojira/arisa/modules/RevokeConfirmationTest.kt
+++ b/src/test/kotlin/io/github/mojira/arisa/modules/RevokeConfirmationTest.kt
@@ -75,8 +75,19 @@ class RevokeConfirmationTest : StringSpec({
     "should return OperationNotNeededModuleResponse when multiple volunteers changed the confirmation status" {
         val module = RevokeConfirmationModule()
         val volunteerChange = ChangeLogItem("Confirmation Status", "Confirmed", Instant.now(), listOf("staff"))
-        val userChange = ChangeLogItem("Confirmation Status", "Unconfirmed", Instant.now(), listOf("helper"))
-        val request = Request("Unconfirmed", listOf(volunteerChange, userChange)) { Unit.right() }
+        val otherVolunteerChange = ChangeLogItem("Confirmation Status", "Unconfirmed", Instant.now(), listOf("helper"))
+        val request = Request("Unconfirmed", listOf(volunteerChange, otherVolunteerChange)) { Unit.right() }
+
+        val result = module(request)
+
+        result.shouldBeLeft(OperationNotNeededModuleResponse)
+    }
+
+    "should return OperationNotNeededModuleResponse when ticket is Unconfirmed and Confirmation Status was unset" {
+        val module = RevokeConfirmationModule()
+        val volunteerChange = ChangeLogItem("Confirmation Status", "Confirmed", Instant.now(), listOf("staff"))
+        val otherVolunteerChange = ChangeLogItem("Confirmation Status", "", Instant.now(), listOf("helper"))
+        val request = Request("Unconfirmed", listOf(volunteerChange, otherVolunteerChange)) { Unit.right() }
 
         val result = module(request)
 
@@ -113,7 +124,22 @@ class RevokeConfirmationTest : StringSpec({
 
         val module = RevokeConfirmationModule()
         val changeLogItem = ChangeLogItem("Confirmation Status", "Confirmed", Instant.now(), emptyList())
-        val request = Request("Confirmed", listOf(changeLogItem)) { it -> changedConfirmation = it; Unit.right() }
+        val request = Request("Confirmed", listOf(changeLogItem)) { changedConfirmation = it; Unit.right() }
+
+        val result = module(request)
+
+        result.shouldBeRight(ModuleResponse)
+        changedConfirmation.shouldBe("Unconfirmed")
+    }
+
+    "should set to Unconfirmed when last volunteer action was setting the confirmation status to empty, and a user changed the confirmation afterwards" {
+        var changedConfirmation = ""
+
+        val module = RevokeConfirmationModule()
+        val volunteerChange = ChangeLogItem("Confirmation Status", "Confirmed", Instant.now(), listOf("staff"))
+        val otherVolunteerChange = ChangeLogItem("Confirmation Status", "", Instant.now(), listOf("helper"))
+        val userChange = ChangeLogItem("Confirmation Status", "Confirmed", Instant.now(), emptyList())
+        val request = Request("Confirmed", listOf(volunteerChange, otherVolunteerChange, userChange)) { changedConfirmation = it; Unit.right() }
 
         val result = module(request)
 

--- a/src/test/kotlin/io/github/mojira/arisa/modules/RevokeConfirmationTest.kt
+++ b/src/test/kotlin/io/github/mojira/arisa/modules/RevokeConfirmationTest.kt
@@ -94,6 +94,28 @@ class RevokeConfirmationTest : StringSpec({
         result.shouldBeLeft(OperationNotNeededModuleResponse)
     }
 
+    "should return OperationNotNeededModuleResponse when confirmation status is null and was unset" {
+        val module = RevokeConfirmationModule()
+        val volunteerChange = ChangeLogItem("Confirmation Status", "Confirmed", Instant.now(), listOf("staff"))
+        val otherVolunteerChange = ChangeLogItem("Confirmation Status", "", Instant.now(), listOf("helper"))
+        val request = Request(null, listOf(volunteerChange, otherVolunteerChange)) { Unit.right() }
+
+        val result = module(request)
+
+        result.shouldBeLeft(OperationNotNeededModuleResponse)
+    }
+
+    "should return OperationNotNeededModuleResponse when confirmation status is empty and was unset" {
+        val module = RevokeConfirmationModule()
+        val volunteerChange = ChangeLogItem("Confirmation Status", "Confirmed", Instant.now(), listOf("staff"))
+        val otherVolunteerChange = ChangeLogItem("Confirmation Status", "", Instant.now(), listOf("helper"))
+        val request = Request("", listOf(volunteerChange, otherVolunteerChange)) { Unit.right() }
+
+        val result = module(request)
+
+        result.shouldBeLeft(OperationNotNeededModuleResponse)
+    }
+
     "should set to Unconfirmed when ticket was created Confirmed" {
         var changedConfirmation = ""
 
@@ -125,21 +147,6 @@ class RevokeConfirmationTest : StringSpec({
         val module = RevokeConfirmationModule()
         val changeLogItem = ChangeLogItem("Confirmation Status", "Confirmed", Instant.now(), emptyList())
         val request = Request("Confirmed", listOf(changeLogItem)) { changedConfirmation = it; Unit.right() }
-
-        val result = module(request)
-
-        result.shouldBeRight(ModuleResponse)
-        changedConfirmation.shouldBe("Unconfirmed")
-    }
-
-    "should set to Unconfirmed when last volunteer action was setting the confirmation status to empty, and a user changed the confirmation afterwards" {
-        var changedConfirmation = ""
-
-        val module = RevokeConfirmationModule()
-        val volunteerChange = ChangeLogItem("Confirmation Status", "Confirmed", Instant.now(), listOf("staff"))
-        val otherVolunteerChange = ChangeLogItem("Confirmation Status", "", Instant.now(), listOf("helper"))
-        val userChange = ChangeLogItem("Confirmation Status", "Confirmed", Instant.now(), emptyList())
-        val request = Request("Confirmed", listOf(volunteerChange, otherVolunteerChange, userChange)) { changedConfirmation = it; Unit.right() }
 
         val result = module(request)
 

